### ISSUE TITLE
Stop working frequently

### ISSUE THREAD

**Issue by menon92** [sarim/ibus-avro#200]
Hi @sarim 
When I try to write using avro it suddenly stops working. I'm using it in Ubuntu 20.04 LTS

I have run `ibus-daemon -xrv` and tried to type using  `ibus avro` after typing some character I'm getting the below error message 
```
IBUS:ERROR:ibuslookuptable.c:323:ibus_lookup_table_set_cursor_pos: assertion failed: (cursor_pos < table->candidates->len)
Bail out! IBUS:ERROR:ibuslookuptable.c:323:ibus_lookup_table_set_cursor_pos: assertion failed: (cursor_pos < table->candidates->len)
```

If I run the `ibus restart` command in my terminal it starts working again. But the same problem comes again after typing some characters

- Thanks

**Comment by sarim** [comment]
Is it random or you can type specific words to trigger it? If so please share those words with `.candidate-selections.json` file from your home folder?

**Comment by menon92** [comment]
It's occurring randomly. Here is an example
![Peek 2023-03-08 09-28](https://user-images.githubusercontent.com/11752205/223612306-c8a7c17e-c9e5-4339-992b-ceab8649e752.gif)

**Comment by menon92** [comment]
Hi, @apandada1,
It did not work for my case. I have already installed a language package

- `OS: Ubuntu 20.04.5 LTS x86_64`
- `DE: GNOME 3.36.5`
- `IBus 1.5.22`

![lang-support](https://user-images.githubusercontent.com/11752205/223904950-cfe5cb20-898a-4726-922d-d9be3790a7c7.png)

Please let me know If you need any other information

`Thanks`

**Comment by menon92** [comment]
Hi, @gunnarhj, Thanks for your response. I have tried the command mentioned in that thread to restart ibus command specifically for GNOME DE.
```sh
$ ps x | grep ibus
  16712 ?        Ssl    0:35 ibus-daemon -rxd --panel disable
  16748 ?        Sl     0:00 /usr/libexec/ibus-dconf
  16752 ?        Sl     0:10 /usr/libexec/ibus-extension-gtk3
  16756 ?        Sl     0:00 /usr/libexec/ibus-x11 --kill-daemon
  16775 ?        Sl     0:00 /usr/libexec/ibus-portal
  29576 ?        Sl     0:08 /usr/libexec/ibus-engine-simple
 119947 pts/1    S+     0:00 grep --color=auto ibus
$ env | grep -E 'XMOD|_IM'
QT_IM_MODULE=ibus
XMODIFIERS=@im=ibus
```
Is there anything that I'm missing?

Thanks

**Comment by gunnarhj** [comment]
@menon92: No, that LGTM. Well, I don't see the ibus-avro process, but it will be there as soon as you switch to Avro.

So the input problem you are experiencing seems not to be caused by a bad IBus configuration.

**Comment by sarim** [comment]
I'm still confused why `jok` is becoming `ho` automatically in his pc (as shown in screencap). That kinda feels some something other than avro is also active? 
@menon92 There is a hidden `.candidate-selections.json` file in your home folder. Rename it to keep backup. Then try again, see if that solves the issue.

**Comment by menon92** [comment]
@sarim I have try remove the `.candidate-selections.json` still facing the same issue. 
Here is today what I have tried. Please let me know If any other test I should perform
![Peek 2023-03-10 09-17](https://user-images.githubusercontent.com/11752205/224214718-6a1b8247-a22d-4a13-9d36-f23e3983dfee.gif)

Thanks
